Unit-Testing-and-Nunit/Collections/Collections_Tests/Collections_Tests.cs

```diff
@@ -0,0 +1,255 @@
+using Collections;
+using NUnit.Framework;
+using System;
+using System.Linq;
+
+namespace Collections_Tests
+{
+    public class CollectionTests
+    {
+
+        [Test]
+        public void Test_EmptyConstructor()
+        {
+            var nums = new Collection<int>();
+
+            Assert.That(nums.Count == 0);
+            Assert.AreEqual(16, nums.Capacity);
+            
+        }
+
+        [Test]
+        [Timeout(1000)]
+        public void Test_OneMillionItems()
+        {
+            int items = 1000000;
+            Collection<int> collection = new Collection<int>();
+            collection.AddRange(Enumerable.Range(0, items).ToArray());
+            Assert.That(collection.Count, Is.EqualTo(items), "Test Collection count should be equal to 1000000");
+            Assert.That(collection.Capacity, Is.GreaterThanOrEqualTo(items), "Test Collection capacity should be greater or equal to 1000000");
+            collection.Clear();
+            Assert.AreEqual(collection.ToString(), "[]");
+            Assert.GreaterOrEqual(collection.Capacity, collection.Count);
+        }
+
+        [Test]
+        public void Test_Collection_ConstructorSingleItem()
+        {
+            var nums = new Collection<int>(20);
+            Assert.That(nums.ToString(), Is.EqualTo("[20]"));
+            Assert.That(nums.Count == 1);
+            Assert.AreEqual(1, nums.Count);

```

Comment:
The first two assert are enough for that test. Others asserts do the same thing like first two.You can remove them.